File: 프로그래머스/lv1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        String answer = "";
        char[] chars = s.toCharArray();

        int idx = 1;
        for (char c : chars) {
            if (!Character.isAlphabetic(c)) {
                idx = 1;
                answer += c;
                continue;
            }

            if (idx % 2 != 0) answer += Character.toUpperCase(c);
            else answer += Character.toLowerCase(c);
            
            idx++;
        }

        return answer;
    }
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    return s.split(" ").map(str => {
        const strArr = str.split("");
        for(let i = 0; i < strArr.length; i++) {
            strArr[i] = i%2==0 ? strArr[i].toUpperCase() : strArr[i].toLowerCase();
        }
        return strArr.join("");
    }).join(" ");;
}